@here/harp-texture-atlas-generator/src/ImageVectorCoders.ts

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Jimp from "jimp";
+import { ColorUtils } from "./ColorUtils";
+import { FileSystem, ImageFormat } from "./FileSystem";
+import { ImageBitmapEncoderConstructor } from "./ImageBitmapCoders";
+import { Color, ImageDecoder, ImageDecoderConstructor } from "./ImageFactory";
+// tslint:disable-next-line: no-duplicate-imports
+import { ImageEncoder, ImageEncoderConstructor } from "./ImageFactory";
+
+// No @types for this module
+// tslint:disable:no-var-requires
+const svg2png = require("svg2png");
+
+export class ImageVectorDecoderConstructor implements ImageDecoderConstructor {
+    readonly targetWidth: number;
+    readonly targetHeight: number;
+
+    constructor(desiredWidth: number, desiredHeight: number) {
+        this.targetWidth = desiredWidth;
+        this.targetHeight = desiredHeight;
+    }
+
+    load(filePath: string): Promise<ImageDecoder> {
+        return new Promise((resolve, reject) => {
+            const loadPromise = FileSystem.readFile(filePath);
+            loadPromise
+                .then(
+                    (fileBuffer: Buffer) => {
+                        svg2png(fileBuffer,
+                            this.targetWidth !== 0 && this.targetHeight !==0 ?
+                            { width: this.targetWidth, height: this.targetHeight } :
+                            {})
+                            .then((pngBuffer: Buffer) => {
+                                Jimp.read(pngBuffer)
+                                    .then((bitmap: Jimp) => {
+                                        resolve(new ImageVectorDecoder(fileBuffer, bitmap));
+                                    });
+                            });
+                    }
+                )
+                .catch((error : any) => {
+                    reject(new Error(error));
```

Comment:
Is this needed because for example `svg2png` throws some garbage ?

Reply:
Refactored for async/await syntax.